cmd/cluster-agent/app/app.go

```diff
@@ -197,15 +197,23 @@ func start(cmd *cobra.Command, args []string) error {
 	// start the autoconfig, this will immediately run any configured check
 	common.StartAutoConfig()
 
-	// Start the cluster-check discovery if configured
-	clusterCheckHandler := setupClusterCheck(mainCtx)
-	// start the cmd HTTPS server
-	sc := clusteragent.ServerContext{
-		ClusterCheckHandler: clusterCheckHandler,
-	}
-	if err = api.StartServer(sc); err != nil {
-		return log.Errorf("Error while starting api server, exiting: %v", err)
+	if config.Datadog.GetBool("cluster_checks.enabled") {
+		// Start the cluster check Autodiscovery
+		clusterCheckHandler, err := setupClusterCheck(mainCtx)
+		if err != nil {
+			log.Errorf("Error while setting up cluster check Autodiscovery %v", err)
+		}
+		// Start the cmd HTTPS server
+		sc := clusteragent.ServerContext{
+			ClusterCheckHandler: clusterCheckHandler,
+		}
+		if err = api.StartServer(sc); err != nil {
+			return log.Errorf("Error while starting api server, exiting: %v", err)
```

Comment:
just for clarity I think we should talk about "agent API" here not "api server" WDYT?